src/Sylius/Component/Promotion/Provider/PreQualifiedPromotionsProviderInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Promotion\Provider;
+
+use Sylius\Component\Promotion\Model\PromotionInterface;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+interface PreQualifiedPromotionsProviderInterface
```

Comment:
IMHO that's an implementation detail, pre-qualified says nothing specific to me, but these names do:

- `ActivePromotionsProvider`
- `ActivePromotionsByChannelProvider`

If I want to, I can even create a `PromotionsProvider` that provides all promotions (so it isn't pre-qualified at all). It still fulfills Liskov Substitution Principle and is perfectly correct promotions provider.